invenio_drafts_resources/records/api.py

```diff
@@ -93,12 +124,90 @@ def register(self):
 class Draft(Record):
     """Draft base API for metadata creation and manipulation."""
 
-    # WHY: We want to force the model_cls to be specified by the user
-    # No default one is given, only the base.
+    #
+    # Configuration to be set by a subclass
+    #
+
+    #: The record's SQLAlchemy model class. Must be set by the subclass.
     model_cls = None
+    #: The parent state's SQLAlchemy model class. Must be set by the subclass.
+    versions_model_cls = None
+    #: The parent record's API class. Must be set by the subclass.
+    parent_record_cls = None
 
+    #
+    # System fields
+    #
+
+    #: The internal persistent identifier. Records and drafts share UUID.
     pid = PIDField('id', provider=DraftRecordIdProviderV2, delete=False)
 
+    #: The parent record - the draft is responsible for creating the parent.
+    parent = ParentField(
+        ParentRecord, create=True, soft_delete=False, hard_delete=True)
+
+    #: Version relationship
+    versions = VersionsField(create=True, set_next=True)
+
+    #: The expiry date of the draft.
     expires_at = ModelField()
 
+    #: Revision id of record from which this draft was created.
     fork_version_id = ModelField()
+
+    @classmethod
+    def new_version(cls, record):
+        """Create a draft for a new version of a record.
+
+        The caller is responsible for:
+        1) checking if a draft for a new version already exists
```

Comment:
Should we explicitly address our discussion about creating a new version only for most recent version record *for now* (in the future we want to allow the version switching) here?

(Probably technical hurdles I am not seeing: Would it make sense to (eventually perhaps) have the same mechanism as for above `draft.is_published` but for latest version here and so have 1) be taken care of. This would be consistent with what `publish` does and what `edit` does. Don't know if that is possible yet though.)

Reply:
Yeah, I was not sure what to do here.

For reference here's the part of the service code:
```
# Draft for new version already exists? if so return it
if record.versions.next_draft_id:
    next_draft = self.draft_cls.get_record(
        record.versions.next_draft_id)
    return self.result_item(
        self, identity, next_draft, links_config=links_config)

# Draft for new version does not exists, so create it
next_draft = self.draft_cls.new_version(record)

# Get the latest published record if it's not the current one.
if not record.versions.is_latest:
    record = self.record_cls.get_record(record.versions.latest_id)
```

The tricky part here is to know that the draft already exists, and return it immediately (this part of the code aligns with ``service.edit()``.

Now seeing it again, I honestly think the method works with any record (not only the latest), so probably the comment is actually not relevant.